### PR TITLE
chore: migrate to SDK v0.46

### PR DESCRIPTION
**Pull request by facundomedica**

### CODE REVIEW

File: module/app/app.go [hunk 1]
```diff
@@ -890,10 +896,9 @@ func (app *Gravity) SimulationManager() *module.SimulationManager {
 // API server.
 func (app *Gravity) RegisterAPIRoutes(apiSvr *api.Server, apiConfig config.APIConfig) {
 	clientCtx := apiSvr.ClientCtx
-	rpc.RegisterRoutes(clientCtx, apiSvr.Router)
-	authrest.RegisterTxRoutes(clientCtx, apiSvr.Router)
+	// rpc.RegisterRoutes(clientCtx, apiSvr.Router)
 	authtx.RegisterGRPCGatewayRoutes(clientCtx, apiSvr.GRPCGatewayRouter)
-	ModuleBasics.RegisterRESTRoutes(clientCtx, apiSvr.Router)
+	// ModuleBasics.RegisterRESTRoutes(clientCtx, apiSvr.Router)
```
Comment (facundomedica) [inline module/app/app.go:901]: TODO: Should I replace these with something else?

File: module/app/encoding.go [hunk 2]
```diff
@@ -3,17 +3,13 @@ package app
...
-	gravityparams "github.com/Gravity-Bridge/Gravity-Bridge/module/app/params"
+	gravityparams "github.com/umee-network/Gravity-Bridge/module/app/params"
 )
 
 // MakeEncodingConfig creates an EncodingConfig for gravity.
 func MakeEncodingConfig() gravityparams.EncodingConfig {
 	encodingConfig := gravityparams.MakeEncodingConfig()
-	ethermintcodec.RegisterLegacyAminoCodec(encodingConfig.Amino)
-	ethermintcryptocodec.RegisterInterfaces(encodingConfig.InterfaceRegistry)
+	std.RegisterLegacyAminoCodec(encodingConfig.Amino)
```
Comment (facundomedica) [inline module/app/encoding.go:12]: I guess we are not supporting auto-forward to Evmos for now 😅

File: module/cmd/gravity/cmd/root.go [hunk 3]
```diff
@@ -49,8 +49,8 @@ func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
 		WithInput(os.Stdin).
 		WithAccountRetriever(types.AccountRetriever{}).
 		WithBroadcastMode(flags.BroadcastBlock).
-		WithHomeDir(app.DefaultNodeHome).
-		WithKeyringOptions(ethermint.EthSecp256k1Option())
+		WithHomeDir(app.DefaultNodeHome)
+		// WithKeyringOptions(ethermint.EthSecp256k1Option())
```
Comment (facundomedica) [inline module/cmd/gravity/cmd/root.go:53]: no evmos srry

File: module/x/gravity/client/rest/query.go [hunk 4]
```diff
@@ -1,214 +1,214 @@
 package rest
 
-import (
```
Comment (facundomedica) [inline module/x/gravity/client/rest/query.go:3]: TODO: Figure out if we have to do something with this file

File: module/x/gravity/client/rest/rest.go [hunk 5]
```diff
@@ -27,40 +32,40 @@ const (
 // RegisterRoutes - Central function to define routes that get registered by the main application
 func RegisterRoutes(cliCtx client.Context, r *mux.Router, storeName string) {
```
Comment (facundomedica) [inline module/x/gravity/client/rest/rest.go:33]: TODO: Everything inside this function was commented. Gotta check what should replace it

File: module/x/gravity/keeper/governance_proposals.go [hunk 6]
```diff
@@ -24,17 +24,20 @@ func RegisterProposalTypes() {
 	metadata := "gravity/IBCMetadata"
 	if !govtypes.IsValidProposalType(strings.TrimPrefix(metadata, prefix)) {
 		govtypes.RegisterProposalType(types.ProposalTypeIBCMetadata)
-		govtypes.RegisterProposalTypeCodec(&types.IBCMetadataProposal{}, metadata)
+		// TODO: Migration to v0.46
```
Comment (facundomedica) [inline module/x/gravity/keeper/governance_proposals.go:27]: Some TODOs here, all relating to the same deprecated function `RegisterProposalTypeCodec`.

File: module/x/gravity/keeper/test_common.go [hunk 7]
```diff
@@ -536,31 +541,41 @@ func CreateTestEnv(t *testing.T) TestInput {
...
+	// router := baseapp.NewRouter()
+	// router.AddRoute(bank.AppModule{
+	// 	AppModuleBasic: bank.AppModuleBasic{},
+	// }.Route())
+	// router.AddRoute(staking.AppModule{
+	// 	AppModuleBasic: staking.AppModuleBasic{},
+	// }.Route())
+	// router.AddRoute(distribution.AppModule{
+	// 	AppModuleBasic: distribution.AppModuleBasic{},
+	// }.Route())
```
Comment (facundomedica) [inline module/x/gravity/keeper/test_common.go:553]: TODO: And this?

File: module/app/app.go [hunk 8]
```diff
@@ -891,10 +896,9 @@ func (app *Gravity) SimulationManager() *module.SimulationManager {
...
```
Comment (facundomedica) [inline module/app/app.go:901]: TODO What should we do here?

File: docs/developer/environment-setup.md [hunk 9]
```diff
@@ -10,9 +10,9 @@ We're always happy to help new developers get started
 
 Gravity bridge has three major components
 
-[The Gravity bridge Solidity](https://github.com/Gravity-Bridge/Gravity-Bridge/tree/main/solidity) and associated tooling. This requires NodeJs
-[The Gravity bridge Cosmos Module and test chain](https://github.com/Gravity-Bridge/Gravity-Bridge/tree/main/module). this requires Go.
-[The Gravity bridge tools](https://github.com/Gravity-Bridge/Gravity-Bridge/tree/main/orchestrator) these require Rust.
+[The Gravity bridge Solidity](https://github.com/umee-network/Gravity-Bridge/tree/main/solidity) and associated tooling. This requires NodeJs
```
Comment (robert-zaremba) [inline docs/developer/environment-setup.md:13]: I don't think we should update this URLs. Ideally these changes will go upstream.

**Review by robert-zaremba — COMMENTED** [review]
I have https://github.com/umee-network/Gravity-Bridge/tree/v1.6-sdk-046 branch with updates covering other dependencies and tests and app migration.

Why creating another branch?

File: module/app/app.go (hunk 8 above)
Comment (robert-zaremba) [reply module/app/app.go:901]: REST routes are deprecated since 0.45. See changes in my branch.

**Comment by robert-zaremba** [comment]
I suggest to continue the work in https://github.com/umee-network/Gravity-Bridge/tree/v1.6-sdk-046 without updating the links.

**Pull request closed without merging**